File: user-srv/src/Pi.User/Pi.User.Domain/AggregatesModel/UserInfoAggregate/NotificationPreference.cs
using Pi.Common.SeedWork;
using Pi.User.Domain.SeedWork;

namespace Pi.User.Domain.AggregatesModel.UserInfoAggregate;

public class NotificationPreference : Entity, IAuditableEntity
{
    public NotificationPreference(bool important, bool order, bool portfolio, bool wallet,
        bool market)
    {
        Important = important;
        Order = order;
        Portfolio = portfolio;
        Wallet = wallet;
        Market = market;
    }

    public Guid Id { get; private set; }
    public bool Important { get; private set; }
    public bool Order { get; private set; }
    public bool Portfolio { get; private set; }
    public bool Wallet { get; private set; }
    public bool Market { get; private set; }
    public Guid DeviceForeignKey { get; }
    public Guid? UserInfoId { get; }
    public UserInfo? UserInfo { get; }

    public void Update(
        bool? important,
        bool? order,
        bool? portfolio,
        bool? wallet,
        bool? market)
    {
        if (important != null)
        {
            Important = (bool)important;
        }

        if (order != null)
        {
            Order = (bool)order;
        }

        if (portfolio != null)
        {
            Portfolio = (bool)portfolio;
        }

        if (wallet != null)
        {
            Wallet = (bool)wallet;
        }

        if (market != null)
        {
            Market = (bool)market;
        }
    }

    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
}

File: user-srv-v2/internal/domain/notification_preference.go
package domain

type NotificationPreference struct {
	Id               string  `gorm:"column:id;type:varchar(36);primaryKey" json:"id"`
	Important        bool    `gorm:"column:important" json:"important"`
	Order            bool    `gorm:"column:order" json:"order"`
	Portfolio        bool    `gorm:"column:portfolio" json:"portfolio"`
	Wallet           bool    `gorm:"column:wallet" json:"wallet"`
	Market           bool    `gorm:"column:market" json:"market"`
	DeviceForeignKey string  `gorm:"column:device_foreign_key;type:varchar(36)" json:"deviceForeignKey"`
	UserInfoId       *string `gorm:"column:user_info_id;type:varchar(36)" json:"userInfoId"`
}
